renku/core/utils/metadata.py

```diff
@@ -101,6 +107,42 @@ def get_modified_activities(
     return modified, deleted
 
 
+def filter_overridden_activities(activities: List["Activity"]) -> List["Activity"]:
+    """Filter out overridden activities from a list of activities."""
+    relevant_activities = {}
+
+    for activity in activities[::-1]:
+        outputs = frozenset(g.entity.path for g in activity.generations)
+
+        subset_of = []
+        superset_of = []
+
+        for k, a in relevant_activities.items():
+            if outputs.issubset(k):
+                subset_of.append((k, a))
+            elif outputs.issuperset(k):
+                superset_of.append((k, a))
+
+        if not subset_of and not superset_of:
+            relevant_activities[outputs] = activity
+            continue
+
+        subset_of = [(k, a) for k, a in relevant_activities.items() if outputs.issubset(k) or outputs.issuperset(k)]
```

Comment:
This is not needed I believe since we already calculated the correct `subset_of` before.